src/hotspot/share/code/vmreg.cpp

```diff
@@ -30,7 +30,7 @@
 // used by SA and jvmti, but it's a leaky abstraction: SA and jvmti
 // "know" that stack0 is an integer masquerading as a pointer. For the
 // sake of those clients, we preserve this interface.
-VMReg VMRegImpl::stack0 = (VMReg)VMRegImpl::stack_0()->value();
+VMReg VMRegImpl::stack0 = (VMReg)(intptr_t)VMRegImpl::stack_0()->value();
```

Comment:
So enlighten me - why the two-step cast?

Reply:
I changed value() to return an int rather than intptr_t, so the compiler objects:

src/hotspot/share/code/vmreg.cpp:33:27: error: cast to pointer from integer of different size [-Werror=int-to-pointer-cast]
   33 | VMReg VMRegImpl::stack0 = (VMReg)VMRegImpl::stack_0()->value();
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~